### PR TITLE
IEP-813 Debugger: Update the hardware-watchpoint-limit based on the selection of the target

### PR DESCRIPTION
**Pull request by sigmaaa**
## Description

The default command for setting hardware-watchpoint-limit was changed in a way, that allows us to adjust it based on the target selection. For example, esp32 has 2 breakpoints/watchpoints and esp32-c3 has 8 hardware breakpoints/watchpoints

Fixes # ([IEP-813](https://jira.espressif.com:8443/browse/IEP-813))

## Type of change

Please delete options that are not relevant.
- New feature (non-breaking change which adds functionality)

## How has this been tested?

Test 1:
- Create a new debug configuration
- edit the new configuration, go to the Startup tab. The new initialization commands line should look like this: 
Screenshot 2023-03-28 at 19 34 35

Test 2:
- Debug the project with different boards. It should be a different number of hardware watchpoints based on the built idf target. The numbers are next: esp32="2", esp32s2="2", esp32s3="2", esp32c3="8", esp32c2="2", "esp32c6"="4", "esp32h2"="8".
- You can check the command in the log after debugging. It will be a message like this: `!MESSAGE Initialization Commands:: mon reset halt
flushregs
set remote hardware-watchpoint-limit 2`

**Test Configuration**:
* ESP-IDF Version:
* OS (Windows,Linux and macOS):

## Dependent components impacted by this PR:

- Startup Tab (Initialization Commands)
- Debug

## Checklist
- [ ] PR Self Reviewed
- [ ] Applied Code formatting
- [ ] Added Documentation
- [ ] Added Unit Test
- [ ] Verified on all platforms - Windows,Linux and macOS

### CODE REVIEW

**Review by kolipakakondal — APPROVED** [review]
Thanks, @sigmaaa for the PR. LGTM

File: bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/DebuggerCommands.java [hunk 1]
```diff
@@ -124,11 +131,14 @@ public IStatus addFirstResetCommands(List<String> commandsList) {
...
-		
+
+		otherInits = otherInits.replace(DefaultPreferences.IDF_TARGET_CPU_WATCHPOINT_NUM,
+				String.valueOf(IdfHardwareWatchpoints.getWatchpointNumForTarget(getCurrentTatget())));
 		otherInits = DebugUtils.resolveAll(otherInits, fAttributes);
 		if (fDoDoubleBackslash && EclipseUtils.isWindows()) {
 			otherInits = StringUtils.duplicateBackslashes(otherInits);
 		}
+		Logger.log("Initialization Commanands::" + otherInits); //$NON-NLS-1$
 		DebugUtils.addMultiLine(otherInits, commandsList);
```
[the remark this answers is not in the document]
Comment (sigmaaa) [reply bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/DebuggerCommands.java:151]: fixed

File: bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/DebuggerCommands.java [hunk 2]
```diff
@@ -141,6 +151,28 @@ public IStatus addFirstResetCommands(List<String> commandsList) {
...
+			}
+			project = activeConfig.getMappedResources()[0].getProject();
+			Logger.log("Project:: " + project); //$NON-NLS-1$
+		}
+		catch (CoreException e)
+		{
+			Logger.log(e);
+		}
+		return new SDKConfigJsonReader(project).getValue("IDF_TARGET"); //$NON-NLS-1$
+	}
```
[the remark this answers is not in the document]
Comment (sigmaaa) [reply bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/DebuggerCommands.java]: done

File: bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/DebuggerCommands.java [hunk 3]
```diff
@@ -141,6 +151,28 @@ public IStatus addFirstResetCommands(List<String> commandsList) {
 		return Status.OK_STATUS;
 	}
 
+	private String getCurrentTatget()
```
[the remark this answers is not in the document]
Comment (sigmaaa) [reply bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/DebuggerCommands.java]: fixed

**Comment by sigmaaa** [comment]
Hi @kolipakakondal, thanks for the review. Fixed mentioned issues in the last commit

**Review by alirana01 — COMMENTED** [review]
LGTM

**Comment by AndriiFilippov** [comment]
Hi @sigmaaa ! 

Tested: 
OS - Windows 10 
ESP-IDF: v5.0.1 

LGTM. 
Able to see new initialization command

image


and the same info in workspace log file: 

image

**Review by kolipakakondal — APPROVED** [review]
LGTM

**Pull request merged**